src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -323,6 +327,31 @@ public ElementInfos getBranchOrThreeWindingsTransformer(UUID networkUuid, String
         throw new ResponseStatusException(HttpStatus.NO_CONTENT);
     }
 
+    private static List<SelectedShuntCompensatorData> toShuntCompensatorData(String lccBusOrBusbarSectionId, Stream<ShuntCompensator> shuntCompensators) {
+        return shuntCompensators
+                .map(s -> SelectedShuntCompensatorData.builder()
+                        .id(s.getId())
+                        .selected(Objects.equals(lccBusOrBusbarSectionId, getBusOrBusbarSection(s.getTerminal())))
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    public HvdcShuntCompensatorInfos getHvcLineWithShuntCompensators(UUID networkUuid, String variantId, String hvdcId) {
+        HvdcShuntCompensatorInfos.HvdcShuntCompensatorInfosBuilder builder = HvdcShuntCompensatorInfos.builder().id(hvdcId);
+        Network network = getNetwork(networkUuid, PreloadingStrategy.NONE, variantId);
```

Comment:
Maybe PreloadingStrategy.COLLECTION, as we can retrieve also multiple shunt compensators ...
But it's not very important ...

Reply:
not sure it worths it - to be checked